perceval/components/abstract_processor.py

```diff
@@ -560,10 +561,10 @@ def check_input(self, input_state: BasicState):
             f"Input length not compatible with circuit (expects {expected_input_length}, got {len(input_state)})"
 
     def _deduce_min_detected_photons(self, expected_photons: int) -> None:
-        warnings.warn(UserWarning(
-            f"Setting a value for min_detected_photons will soon be mandatory, please change your scripts accordingly\n" +
-            "Use the method processor.min_detected_photons_filter(value) before any call of processor.with_input(input)\n" +
-            "The current deduced value of min_detected_photons is {expected_photons}"))
+        logger.warn(
+            "Setting a value for min_detected_photons will soon be mandatory, please change your scripts accordingly." +
+            " Use the method processor.min_detected_photons_filter(value) before any call of processor.with_input(input)." +
+            f" The current deduced value of min_detected_photons is {expected_photons}", channel.user)
```

Comment:
We don't want any warning in case of perfect source (no noise) in local simulation, so it's normal. As for the two warnings with local processor and an imperfect source and not setting min_detected_photons_filter, it's also the expected behavior :)